File: abc/abc146/abc146_c.cs
// https://atcoder.jp/contests/abc146/submissions/8637354

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(long.Parse).ToArray();
            var A = input[0];
            var B = input[1];
            var X = input[2];

            if (A + B >= X)
            {
                Console.WriteLine(0);
                return;
            }

            if (A * Math.Pow(10, 9) + B * 10 <= X)
            {
                Console.WriteLine(Math.Pow(10, 9));
                return;
            }

            var n = (long)Math.Pow(10, 9);
            var result = 0L;
            while (n > 0L)
            {
                if (A * n + B * n.ToString().Length <= X)
                {
                    result = Math.Min(n * 10 - 1, (X - B * n.ToString().Length) / A);
                    break;
                }
                n /= 10L;
            }

            Console.WriteLine(Math.Max(0, result));
        }
    }
}

File: abc/abc146/abc146_c.fs
// https://atcoder.jp/contests/abc146/submissions/12155746

let [| A; B; X |] = stdin.ReadLine().Split() |> Array.map int64

let rec resolver n =
    if A * n + B * int64 (n.ToString().Length) <= X then min (n * 10L - 1L) ((X - B * int64 (n.ToString().Length)) / A)
    else resolver (n / 10L)

if A + B >= X then 0L
elif A * pown 10L 9 + B * 10L <= X then pown 10L 9
else resolver (pown 10L 9)
|> stdout.WriteLine
